compiler_opt/tools/extract_ir_lib.py

```diff
@@ -134,32 +134,40 @@ def _extract_clang_artifacts(self, llvm_objcopy_path: str, cmd_filter: str,
       logging.info('%s does not exist.', self.input_obj())
       return None
     os.makedirs(self.dest_dir(), exist_ok=True)
-    try:
-      subprocess.run(
-          self._get_extraction_cmd_command(llvm_objcopy_path, cmd_section_name),
-          check=True)
-      if cmd_filter is not None or is_thinlto:
-        with open(self.cmd_file(), encoding='utf-8') as f:
-          lines = f.readlines()
-        assert len(lines) == 1
-        cmdline = lines[0]
-        if not should_include_module(cmdline, cmd_filter):
-          logging.info(
-              'Excluding module %s because it does not match the filter',
-              self.input_obj())
-          os.remove(self.cmd_file())
-          return None
-        if is_thinlto:
-          index_file = get_thinlto_index(cmdline, self.obj_base_dir())
-          shutil.copy(index_file, self.thinlto_index_file())
-
-      subprocess.run(
-          self._get_extraction_bc_command(llvm_objcopy_path,
-                                          bitcode_section_name),
-          check=True)
-    except subprocess.CalledProcessError as e:
-      # This may happen if  .o file was build from asm (.S source).
-      logging.warning('%s was not processed: %s', self.input_obj(), e)
+    cmd_objcopy_output = subprocess.run(
```

Comment:
why not use `check_output` in a try/catch, to avoid explicitly checking the `returncode` (i.e. seems like something that API is for?)

Reply:
Fixed. That is definitely a cleaner solution.